### PR TITLE
Support Keyed DI services in .NET 8

### PR DESCRIPTION
**Pull request by rickdotnet**
PR for Issue #315 (Support Keyed DI services in .NET 8)

### CODE REVIEW

File: src/NATS.Client.Hosting/NatsHostingExtensions.cs [hunk 1]
```diff
@@ -8,11 +8,24 @@ namespace NATS.Client.Hosting;
...
+    /// Others, registered `NatsConnectionPool` as singleton, `NatsConnection` and `INatsConnection` as transient(get from pool).
     /// </summary>
-    public static IServiceCollection AddNats(this IServiceCollection services, int poolSize = 1, Func<NatsOpts, NatsOpts>? configureOpts = null, Action<NatsConnection>? configureConnection = null)
+    public static IServiceCollection AddNats(
+        this IServiceCollection services,
+        int poolSize = 1,
+        Func<NatsOpts, NatsOpts>? configureOpts = null,
+        Action<NatsConnection>? configureConnection = null
+#if NET8_0_OR_GREATER
+        , string? key = null // This parameter is only available in .NET 8 or greater
```
Comment (mtmk) [inline src/NATS.Client.Hosting/NatsHostingExtensions.cs:20]: if you could suppress `Commas should not be preceded by whitespace` warnings we can get the build to be green (an unintentional check in native aot tests breaking the build for *any* earning)

File: src/NATS.Client.Hosting/NATS.Client.Hosting.csproj [hunk 2]
```diff
@@ -13,10 +13,14 @@
     <IsPackable>true</IsPackable>
   </PropertyGroup>
 
-  <ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' != 'net8.0'">
```
Comment (mtmk) [inline src/NATS.Client.Hosting/NATS.Client.Hosting.csproj:16]: nice!

**Review by mtmk — COMMENTED** [review]
LGTM just the build warning needs a tweak. thanks @rickdotnet

File: src/NATS.Client.Hosting/NatsHostingExtensions.cs (hunk 1 above)
Comment (rickdotnet) [reply src/NATS.Client.Hosting/NatsHostingExtensions.cs:24]: I added three more that I saw from the warnings on the build. Looks to have done the trick.

**Review by mtmk — APPROVED** [review]

**Pull request merged**